### ISSUE TITLE
All P105s,are not working anymore after firmware update Sept. 2023

### ISSUE THREAD

**Issue by MarcoEddizzo83** [petretiandrea/home-assistant-tapo-p100#576]
Hi, 

as reported in Issue #468  and as has been requested in the same thead, I report what happen with tapo 105s on home assistant, in september 2023:

**--> Version of the integration**

Tapo controller ver.: Stable v2.10.0.1
Home assistant core 2022.11.2 
Home assistant oS 10.3

**--> Configuration**

Tapo Device Name: P105
Firmware version:
- Not working with HA integration: _1.2.1 Build 230804 Rel. 190922._
- Latest working with HA integration: _1.1.5 Build 230324 Rel. 143956._

**--> Describe the bug**

Hi,
I get back on this issue just to report another problem regarding the firmware update of Tapo SP P105.
More than one week ago I noticed that some of this plugs doesn't been recognized by home assistant even if some else was perfectly working.
I have actually plug 8 P105 and 14 P110.

The P105s that works, during the night are out of power (is placed in cascade with a master P110 that switch off during the night).

Checking the p105 "not working" (in home assistant) inside the TAPO APP i'd noticed that they have onboard firwmare version 1.2.1 Build 230804 Rel. 190922.

The p105 that works have onboard firware 1.1.5 Build 230324 Rel. 143956.
That's because I'd enable the auto-update of the firmware during the night (03.00-05.00 AM) and the one's that works was switched off during that time.

Of course the updated one's is in grey in home assistant. I tried:

To delect the instance of the single plug over home assistant and re-create. I got "INVALID AUTHENTICATION" red warning, of course using TAPO-CONTROLLER Hacs integration in home assistant. I triple verify the ip, username and password (I have a DHCP RESERVATION on router for this plugs, bind with mac address). The mail address doesn't have CAPITALIZED letters.
I tried to delete the single plug from TAPO app, re-register and re-join. They come over with the updated firmware :(
Nothing else to do... Of course the invalid authentication seems to depend to the firmware update... the "old" one still works (I deactivated the auto-firmware-update on all the plugs).

Of course directly on tapo app and with alexa, everything works!

I know that this is an open source and custom integration, and I really appreciate the time spent and the effort dedicated. Accordingly with you time, could you please take a look?

I've some automation with this plug that is not working anymore :(

Between the two report I tried 2 net things:
1) change the dhcp reservation ip for the same device (Deleting before from the app, and re-associate. Got the new ip). Not works.
2) try the same with a P110 plug in order to check if I was wrong on autenthication (wrong password, wrong username). With a P110 everything works perfectly (remove from HA, re-add using IP, username and password).

I would say thank in advance for your eventually answer, kind and consideration.

Please let me know if you need any additional information.

Following the HA VERSION:
Home assistant core 2022.11.2
Home assistant oS 10.3
Tapo controller ver.: Stable v2.10.0.1

![Tapo_controller](https://github.com/petretiandrea/home-assistant-tapo-p100/assets/67417471/4580859e-a494-42d8-a2b1-0c5fdc4e840b)

**Comment by Bullameister** [comment]
It would seem that the p100 are affected as well

**Comment by bustedstr** [comment]
Seems my Tapo LED L900 strips are also having issues, possibly due to the update like you said? When doing the usual "add new device" and appending the IP address and account details, I get a "failed to connect" error

**Comment by mdal21** [comment]
Also my L920s not working after September upgrade (1.1.0 Build 230905 Rel.190143). 
Tapo Controller v.2.10.1
Error returned by HA follows:
---------------------------------------
This error originated from a custom integration.

Logger: custom_components.tapo.common_setup
Source: custom_components/tapo/common_setup.py:45
Integration: tapo_p100 (documentation, issues)
First occurred: October 18, 2023 at 5:01:39 PM (3506 occurrences)
Last logged: 7:49:07 AM

Tapo exception: Returned unknown error_code: 1003
Failed to setup Error tapo exception: Returned unknown error_code: 1003

**Comment by MarvAhl** [comment]
My L920 also no longer works.

**Comment by Cromax81** [comment]
Same with P115

**Comment by andrix988** [comment]
I'm following with the same issue with P100...

**Comment by petretiandrea** [comment]
I think I need the physical device to debug it and find a solution

**Comment by daufinsyd** [comment]
I encounter the same issue with P100 also

**Comment by petretiandrea** [comment]
Because I don't have the device, I need a voluntary which expose its device to internet using ngrok. Giving me the address I can test id, we can establish a tapo test account, with fake username and password. When a fix is found, the device will be deleted from tapo test account.

**Comment by side777** [comment]
I bought some new P100s with current firmware and they do connect in HA but I only get the sensors, no switch. Is this related?

**Comment by andrix988** [comment]
> I'm following with the same issue with P100...

Update of my case: now all works well. Unfortunately I made many modifications in wifi network and to HA, then I'm not able to add useful information. Anyway, with the current latest updates of this integration (v2.11.0) and with the latest firmware of tapo P100 (1.5.5. hardw version: 1.20.0) all works well.

**Comment by petretiandrea** [comment]
> I bought some new P100s with current firmware and they do connect in HA but I only get the sensors, no switch. Is this related?

Refresh page, the entities are added asynchronously

**Comment by mdal21** [comment]
Updated the HA to 2023.11.1 and with the integration v2.11.0 my L920s work as expected. No issues then.

**Comment by PtitBen56** [comment]
I was having the same issue so I decided to delete the devices in HA (5 P110) and add them back.  Now it all works as expected.

**Comment by GSzabados** [comment]
@petretiandrea, Tapo has increased local security for many of the devices rolling out with the new firmware updates.
The new Tapo cameras are using now SHA256 hash instead of the previous MD5 and all communication is encrypted on the local network with a double token in play, plus tracked message counts, etc.

**Comment by vpanichkin** [comment]
Hey guys. Here is probably the solution for [OpenHub](https://github.com/openhab/openhab-addons/issues/15585). Maybe it helps you to adjust as well

**Comment by petretiandrea** [comment]
This integration already implements the new KLAP protocol

**Comment by petretiandrea** [comment]
> @petretiandrea, Tapo has increased local security for many of the devices rolling out with the new firmware updates. The new Tapo cameras are using now SHA256 hash instead of the previous MD5 and all communication is encrypted on the local network with a double token in play, plus tracked message counts, etc.

Tapo cameras are out of scope of this integration

**Comment by petretiandrea** [comment]
> Hi at all.
> 
> Reading the latest comment I do tried to do some change. I update HA OS and Core to the latest versione availabe (11.11). Tried to re join the tapo p105 and receive same error message "INVALID AUTHENTICATION".
> 
> Also a tons of other integration not works (ecovacs, shelly and moreover).
> 
> About the tapo p105 i think that the issue apparently seems indepentent to the OS and the core version.
> 
> Luckly I had 3 or 4 backup of the HA VM so I restore immediately what I had before.
> 
> @petretiandrea i think we are geographically close ;) I you would make some test directly with the device involved you think would be a solution?
> 
> _"I need a voluntary which expose its device to internet using ngrok."_ I've a phisycal firewall in the middle, could be complicated. In case we can write in private. Let me know ;)

Yes it's complicated, but using ngrock it's a one click solution, if you want we can try it

**Comment by GSzabados** [comment]
> Tapo cameras are out of scope of this integration

I see, that you have problems understanding what I wrote to you.

TAPO has introduced stronger encryption for local control on all of their devices. They are introduced with the new firmware popping up recently, release notes are like: _Enhanced device security_ or _Enhanced local communication security_.

The cameras were just an example to show what kind of changes were implemented there. But don't worry, a week or so and your device will have the new firmware offered. The P100, P105 and P110 have already got the new firmware released.
